tests/functional/test_s3.py

```diff
@@ -38,13 +40,81 @@ def test_transfer_methods_injected_to_bucket(self):
                         'upload_file was not injected onto S3 bucket')
         self.assertTrue(hasattr(bucket, 'download_file'),
                         'download_file was not injected onto S3 bucket')
+        self.assertTrue(hasattr(bucket, 'copy'),
+                        'copy was not injected onto S3 bucket')
 
     def test_transfer_methods_injected_to_object(self):
         obj = boto3.resource('s3').Object('my_bucket', 'my_key')
         self.assertTrue(hasattr(obj, 'upload_file'),
                         'upload_file was not injected onto S3 object')
         self.assertTrue(hasattr(obj, 'download_file'),
                         'download_file was not injected onto S3 object')
+        self.assertTrue(hasattr(obj, 'copy'),
+                        'copy was not injected onto S3 object')
+
+
+class TestCopy(unittest.TestCase):
```

Comment:
So I think it is good that we are adding this as a functional test, but it would probably be good to assert the expect params. Honestly, I would be good if we add an integration test for a quick ping for copy() as we already have quick pings for upload and download file on all the of the objects the copy() method gets added to. What do you think?